### ISSUE TITLE
(Feature) Add a Hospitality (and Gastronomy) patch for zones.

### ISSUE THREAD

**Issue by modulozero** [alextd/RimWorld-EnhancementPack#16]
TL;DR: add a patch that makes both these mods show only pawn-tagged areas in their selection dialogs (i.e. the guest/shopping/restaurant zone setup), similarly to how Colony Manager already works.

This might be a bit of a bother to implement, especially as [Hospitality](https://github.com/OrionFive/Hospitality) and [Gastronomy](https://github.com/OrionFive/Gastronomy) are complex and actively developed mods, and those would almost certainly be Harmony patches. On the other hand, I feel like it's probably a bit more appropriate than asking Orion for integration. I'm not 100% sure though, so if you tell me so, I might ask on his side.

As my use of the [Cleaning Priority](https://steamcommunity.com/sharedfiles/filedetails/?id=2018316486) mod results in me having _many_ zones, I rely on this mod to make the usual zone selection widgets manageable. Unfortunately neither Hospitality nor Gastronomy supports the tag, making some of their selection widgets _painful_, especially considering that they are already relatively small.

Also, sorry for not making this a PR - I feel like I should do this myself, but I'm suffering plenty brain fog lately :(
